assignments/pull_requests.go

```diff
@@ -0,0 +1,157 @@
+	body += fmt.Sprintf("| **Total** | | | **%d%%** |\n\n", results.TaskSum(task.LocalName()))
+	body += fmt.Sprintf("Once a total score of %d%% is reached, reviewers are automatically assigned.\n", assignment.GetScoreLimit())
+	return body
+}
+
+// AssignReviewers assigns reviewers to a group repository pull request.
+// It assigns one other group member and one course teacher as reviewers.
+func AssignReviewers(ctx context.Context, sc scm.SCM, db database.Database, course *pb.Course, repo *pb.Repository, pullRequest *pb.PullRequest) error {
+	teacherReviewer, err := getNextTeacherReviewer(db, course)
+	if err != nil {
+		return err
+	}
+	// TODO(espeland): Remember to uncomment when finished testing
+	// studentReviewer, err := getNextStudentReviewer(db, repo.GetGroupID(), pullRequest.GetUserID())
+	// if err != nil {
+	// 	return err
+	// }
+
+	reviewers := []string{
+		teacherReviewer.GetLogin(),
+		// studentReviewer.GetLogin(),
+	}
+	opt := &scm.RequestReviewersOptions{
+		Organization: course.GetOrganizationPath(),
+		Repository:   repo.Name(),
+		Number:       int(pullRequest.GetNumber()),
+		Reviewers:    reviewers,
+	}
+
+	if err := sc.RequestReviewers(ctx, opt); err != nil {
+		return err
+	}
+	// Change pull request stage to review
+	pullRequest.SetReview()
+	return db.UpdatePullRequest(pullRequest)
+}
+
+// getNextReviewer gets the next reviewer from either teacherReviewCounter or studentReviewCounter,
+// based on whoever in total has been assigned to the least amount of pull requests.
+// It is simple, and does not account for how many current review requests any user has.
+//
+// Returns an error if the list of users is empty.
+func getNextReviewer(ID uint64, users []*pb.User, reviewCounter map[uint64]map[uint64]int) (*pb.User, error) {
+	if len(users) == 0 {
+		return nil, errors.New("list of users is empty")
+	}
+	reviewerMap, ok := reviewCounter[ID]
+	if !ok {
+		// If a map does not exist for a given ID we create it,
+		// and return the first user.
+		reviewCounter[ID] = make(map[uint64]int)
+		reviewCounter[ID][users[0].GetID()] = 1
+		return users[0], nil
+	}
+	userWithLowestCount := users[0]
+	lowestCount := reviewerMap[users[0].GetID()]
+	for _, user := range users {
+		count, ok := reviewerMap[user.GetID()]
+		if !ok {
+			// If the user is not present in the review map,
+			// then they are returned as the next reviewer.
+			reviewerMap[user.GetID()] = 1
+			return user, nil
+		}
+		if count < lowestCount {
+			userWithLowestCount = user
+			lowestCount = count
+		}
+	}
+	reviewerMap[userWithLowestCount.GetID()]++
+	return userWithLowestCount, nil
+}
+
+// getNextTeacherReviewer gets the teacher with the least total reviews.
+func getNextTeacherReviewer(db database.Database, course *pb.Course) (*pb.User, error) {
+	teachers, err := db.GetCourseTeachers(course)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get teachers from database: %w", err)
+	}
+	if len(teachers) == 0 {
```

Comment:
Same reasoning as below. A course must always have a teacher, and `db.GetCourseTeachers()` should return an error if it doesn't.